persistence/data_models_json_fields.go

```diff
@@ -91,6 +91,164 @@ func (s *StateExecutionSequenceMapsJson) CompleteNewStateExecution(stateId strin
 	return nil
 }
 
+type StateExecutionWaitingQueuesJson struct {
+	// { state_execution_id_1: [ (queue_name_1, count_1), (queue_name_2, count_2), ... ], ... }
+	StateToCommandsMap map[string][]xdbapi.LocalQueueCommand `json:"stateToCommandsMap"`
+	// { queue_name_1: [dedupId_1, dedupId_2, ...], queue_name_2: [dedup_id, ...], ... }
+	UnconsumedMessageQueueDedupIdsMap map[string][]uuid.UUID `json:"unconsumedMap"`
```

Comment:
How about:
```
UnconsumedLocalQueueMessages map[string][]InternalLocalQueueMessage`json:"unconsumedLocalQueueMessages"`

InternalLocalQueueMessage struct{
    DedupId string
    IsFull bool // only false for now until we support including payload
}
     
```
* So that we can support the optimization later
* JSON has pretty bad support for byte array actually. each byte will consume 5? bytes to store, like \xAAAA.

Reply:
Good idea.